app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaScreenModel.kt

```diff
@@ -537,7 +538,13 @@ class MangaScreenModel(
         val state = successState ?: return
         try {
             withIOContext {
-                val chapters = state.source.getChapterList(state.manga.toSManga())
+                val manga = state.manga.toSManga()
+                val chapters = state.source.getChapterList(manga)
+
+                // `LocalSource.getChapterList` has side effects since 82bdf63
+                if (state.manga != state.manga.copyFrom(manga)) {
+                    updateManga.awaitUpdateFromSource(state.manga, manga, manualFetch)
+                }
```

Comment:
Since this is specific to LocalSource move the code inside it?

Reply:
Yes

Reply:
It's not feasible as LocalSource doesn't (and shouldn't) have access to update manga states, have to do it in MangaScreenModel.